recipes/import-assertions-to-attributes/src/workflow.ts

```diff
@@ -0,0 +1,154 @@
+import { styleText } from 'node:util';
+import { api } from '@codemod.com/workflow';
+import type { Api } from '@codemod.com/workflow';
+import type { Edit } from '@ast-grep/napi';
+
+// We run this codemod on all JS/TS files
+// including CJS, MJS, TS, and JSX files.
+const globPattern = '**/*.{cjs,mjs,js,jsx,?(d.)cts,?(d.)mts,?(d.)ts,tsx}';
+
+async function processModule(astGrep: Api['astGrep'], contexts: Api['contexts']) {
+	await astGrep({
+        rule: {
+            any: [
+				/**
+				 * cover `import { something } from './module.json' assert { type: 'json' };`
+				 *
+				 * We are searching for an ERROR child
+				 * because the import assertions syntax isn't correctly parsed
+				 * by the parser, so we need to handle it manually.
+				 */
+				{
+					kind: 'import_statement',
+					has: {
+						kind: 'ERROR',
+						inside: {
+							kind: 'import_statement'
+						}
+					}
+				},
+				// cover `import('./module.json', { assert: { type: 'json' } })`
+				{
+					kind: 'call_expression',
+					pattern: "import($SPECIFIER, { assert: $ASSERT })",
```

Comment:
I think this can be narrowed down first to just the second argument. You need to drill down something like kind: arguments → kind: object → kind: pair → kind: property_identifier → key → regex: '^assert$'

The benefit of that is reducing the replacement to just the part you actually care about.

Reply:
No because we need to be sure we are modifying the right function